### PR TITLE
update/add: Warns/Bans/Bot Custom Status & More

### PR DESCRIPTION
**Pull request by juddisjudd**
NOTE: When testing I had to kick dyno out of the test server before being able to see the ban and warn commands.
This could have potentially just been a discord caching issue.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = "file:./database.db"
+}
+
+model Warn {
+  id        Int      @id @default(autoincrement())
```
Comment (LukeWasTakenn) [inline prisma/schema.prisma:11]: Minor thing but there's no reason for these to not be unsigned

File: src/commands/ban.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      await prisma.ban.create({
+        data: {
+          reason: reason,
+          issuerId: interaction.user.id,
+          targetId: user.id,
+        },
+      });
+
+      await interaction.guild.members.ban(user, { reason: reason, deleteMessageDays: deleteMessageDays });
+      await interaction.reply({ content: `Banned <@${user.id}>. Reason: ${reason}`, ephemeral: true });
```
Comment (LukeWasTakenn) [inline src/commands/ban.ts]: shouldn't we try to add the entry into the database *after* the bot has banned the user and not thrown an error?

File: src/commands/warn.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  try {
+    const dmChannel = await user.createDM();
+    await dmChannel.send(dmMessage);
+  } catch (err) {
+    console.error('Failed to send DM:', err);
+  }
+}
+
+function calculateTimeoutDuration(warnCount: number): number {
+  return 60000 * warnCount; // 1 minute per warn
```
Comment (LukeWasTakenn) [inline src/commands/warn.ts]: Does discord let you time out users for an amount outside of their specified duration?

Also 1 minute per warn (3 minutes for 3 warns) seems kinda low, maybe start at 5 minutes -> 10 minutes -> 1 hour, etc...?

**Review by LukeWasTakenn — CHANGES_REQUESTED** [review]
I feel like commands should be organized into folders as well, `moderation`, `misc`, etc...

But overall PR looks pretty good

File: src/commands/warn.ts (hunk 3 above)
Comment (ckh-rcit) [reply src/commands/warn.ts:99]: Sure, asked linden and kind of went off what he said.

File: src/commands/ban.ts (hunk 2 above)
Comment (juddisjudd) [reply src/commands/ban.ts]: Done

File: prisma/schema.prisma (hunk 1 above)
Comment (juddisjudd) [reply prisma/schema.prisma:11]: sqlite does not have unsigned integers as a distinct data type

**Review by LukeWasTakenn — APPROVED** [review]
lgtm, thank you

**Pull request merged**